Compiler/src/typelimits.jl

```diff
@@ -638,7 +694,13 @@ end
                             ⋤(𝕃, tyi, ft) # just a type-level information, but more precise than the declared type
             end
         end
-        anyrefine && return PartialStruct(𝕃, aty, fields)
+        if isa(typea, PartialStruct) && isa(typeb, PartialStruct) &&
+            isvarargtype(typea.fields[end]) && isvarargtype(typeb.fields[end])
+            # XXX: If it may be more precise than `Vararg` (e.g. `Vararg{T}`),
+            # handle that in the main loop above to get a more accurate type.
+            push!(fields, Vararg)
+        end
```

Comment:
Not sure if I should be more careful about `Vararg` handling here, seems to be going fine as it is.